src/utils/api_helper.ts

```diff
@@ -1656,3 +1671,194 @@ export const requestInvoice = async (
     payload
   )
 }
+
+export const getWalletTransactions = async (
+  wallet_address: string,
+  token_address?: string,
+  chain_id?: number,
+  limit?: number,
+  offset?: number
+) => {
+  return await internalFetch(`/secure/transactions/wallet`, 'POST', {
+    wallet_address,
+    token_address,
+    chain_id,
+    limit,
+    offset,
+  })
+}
+
+// New Thirdweb-based balance functions
+export const getNativeBalance = async (
+  walletAddress: string,
+  chain: SupportedChain
+): Promise<{ balance: number }> => {
+  try {
+    const chainInfo = getChainInfo(chain)
+    if (!chainInfo) {
+      throw new Error(`Unsupported chain: ${chain}`)
+    }
+
+    const publicClient = viemAdapter.publicClient.toViem({
+      chain: chainInfo.thirdwebChain,
+      client: thirdWebClient,
+    })
+
+    const balance = await publicClient.getBalance({
+      address: walletAddress as `0x${string}`,
+    })
+
+    return { balance: Number(balance) / 1e18 } // Convert from wei to ether
+  } catch (error) {
+    console.error('Error getting native balance:', error)
+    return { balance: 0 }
+  }
+}
+
+export const getTokenBalance = async (
+  walletAddress: string,
+  tokenAddress: string,
+  chain: SupportedChain
+): Promise<{ balance: number }> => {
+  try {
+    const chainInfo = getChainInfo(chain)
+    if (!chainInfo) {
+      throw new Error(`Unsupported chain: ${chain}`)
+    }
+
+    const contract = getContract({
+      client: thirdWebClient,
+      chain: chainInfo.thirdwebChain,
+      address: tokenAddress as `0x${string}`,
+      abi: erc20Abi,
+    })
+
+    const balance = await readContract({
+      contract,
+      method: 'balanceOf',
+      params: [walletAddress as `0x${string}`],
+    })
+
+    // Get decimals for proper formatting
+    const decimals = await readContract({
+      contract,
+      method: 'decimals',
+    })
+
+    return { balance: Number(balance) / Math.pow(10, decimals) }
+  } catch (error) {
+    console.error('Error getting token balance:', error)
+    return { balance: 0 }
+  }
+}
+
+export const getTotalWalletBalance = async (
+  walletAddress: string
+): Promise<{ balance: number }> => {
+  try {
+    const supportedChains = [
+      SupportedChain.CELO,
+      SupportedChain.ARBITRUM,
+      SupportedChain.ARBITRUM_SEPOLIA,
+    ]
+
+    // Get all token configurations from chains.ts
+    const tokenConfigs = [
```

Comment:
Why isn't this being dynamically pulled? All the data defined here already exists on the "supportedChains.acceptableTokens". With this setup, we need to update tokens in two places when the only thing we have changing is the chains.

Reply:
Dynamically pulling the data should also ensure that the token decimal is always in context